File: app/src/main/java/com/appchamp/wordchunks/ui/game/fragments/LevelSolvedFrag.java
package com.appchamp.wordchunks.ui.game.fragments;

import android.content.Context;
import android.content.res.Resources;
import android.graphics.drawable.GradientDrawable;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.appchamp.wordchunks.R;
import com.appchamp.wordchunks.ui.game.listeners.OnNextLevelListener;

import java.util.Random;


public class LevelSolvedFrag extends Fragment {

    private OnNextLevelListener callback;

    private RelativeLayout rlNextLevel;
    private ImageView imgRectBg;
    private TextView tvExcellent;
    private TextView tvFunFact;
    private TextView tvNextLevelTitle;
    private TextView tvNextLevelClue;
    private TextView tvLevelsLeft;

    public LevelSolvedFrag() {
        // Requires empty public constructor
    }

    public static LevelSolvedFrag newInstance() {
        //int color, String clue, String fact, long levelsLeft) {
        return new LevelSolvedFrag();
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View root = inflater.inflate(R.layout.frag_level_solved, container, false);
        rlNextLevel = (RelativeLayout) root.findViewById(R.id.rlNextLevel);
        imgRectBg = (ImageView) root.findViewById(R.id.imgRectBg);
        tvExcellent = (TextView) root.findViewById(R.id.tvExcellent);
        tvFunFact = (TextView) root.findViewById(R.id.tvFunFact);
        tvNextLevelTitle = (TextView) root.findViewById(R.id.tvNextLevelTitle);
        tvNextLevelClue = (TextView) root.findViewById(R.id.tvNextLevelClue);
        tvLevelsLeft = (TextView) root.findViewById(R.id.tvLevelsLeft);

        rlNextLevel.setOnClickListener(v -> callback.onNextLevelSelected());

        setPackColor();
        setClue();
        setFunFact();
        setLevelsLeft();
        setExcellent();

        return root;
    }

    private void setPackColor() {
        int color = getArguments().getInt("color");
        GradientDrawable drawable = (GradientDrawable) imgRectBg.getDrawable();
        drawable.setColor(color);
        tvNextLevelTitle.setTextColor(color);
    }

    private void setClue() {
        String clue = getArguments().getString("clue");
        if (clue == "") {
            tvNextLevelClue.setText("CONGRATULATIONS!");
        } else {
            tvNextLevelClue.setText(clue);
        }
    }

    private void setFunFact() {
        String fact = getArguments().getString("fact");
        tvFunFact.setText(fact);
    }

    private void setLevelsLeft() {
        long left = getArguments().getLong("left");
        if (left == 0) {
            tvLevelsLeft.setText("YOU'VE FINISHED THE WHOLE PACK!");
        } else if (left == 1) {
            tvLevelsLeft.setText("ONLY ONE LEVEL LEFT IN PACK");
        } else {
            tvLevelsLeft.setText(left + " LEVELS LEFT IN PACK");
        }
    }

    private void setExcellent() {
        Resources res = getContext().getResources();
        String[] congrats = res.getStringArray(R.array.congrats);
        int i = new Random().nextInt(congrats.length - 1);
        tvExcellent.setText(congrats[i]);
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        // This makes sure that the container activity has implemented
        // the callback interface. If not, it throws an exception
        try {
            callback = (OnNextLevelListener) context;
        } catch (ClassCastException e) {
            throw new ClassCastException(context.toString()
                    + " must implement OnNextLevelListener");
        }
    }

}

File: app/src/main/java/com/appchamp/wordchunks/ui/game/fragments/LevelSolvedFrag.kt
package com.appchamp.wordchunks.ui.game.fragments

import android.content.Context
import android.graphics.drawable.GradientDrawable
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.appchamp.wordchunks.R
import com.appchamp.wordchunks.ui.game.listeners.OnNextLevelListener
import com.appchamp.wordchunks.util.Constants.CLUE_ID_KEY
import com.appchamp.wordchunks.util.Constants.COLOR_ID_KEY
import com.appchamp.wordchunks.util.Constants.FACT_ID_KEY
import com.appchamp.wordchunks.util.Constants.LEFT_ID_KEY
import kotlinx.android.synthetic.main.frag_level_solved.*
import java.util.*


class LevelSolvedFrag : Fragment() {

    private var callback: OnNextLevelListener? = null

    companion object {

        fun newInstance(color: Int, clue: String, fact: String, left: Long): LevelSolvedFrag {
            val args = Bundle()
            args.putInt(COLOR_ID_KEY, color)
            args.putString(CLUE_ID_KEY, clue)
            args.putString(FACT_ID_KEY, fact)
            args.putLong(LEFT_ID_KEY, left)
            val levelSolvedFrag: LevelSolvedFrag = newInstance()
            levelSolvedFrag.arguments = args
            return levelSolvedFrag
        }

        fun newInstance(): LevelSolvedFrag = LevelSolvedFrag()
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater!!.inflate(R.layout.frag_level_solved, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        rlNextLevel.setOnClickListener { callback!!.onNextLevelSelected() }

        setPackColor()
        setClue()
        setFunFact()
        setLevelsLeft()
        setExcellent()
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        // This makes sure that the container activity has implemented
        // the callback interface. If not, it throws an exception
        try {
            callback = context as OnNextLevelListener?
        } catch (e: ClassCastException) {
            throw ClassCastException(context!!.toString() + " must implement OnNextLevelListener")
        }

    }

    private fun setPackColor() {
        val color = arguments.getInt(COLOR_ID_KEY)
        val drawable = imgRectBg.drawable as GradientDrawable
        drawable.setColor(color)
        tvNextLevelTitle.setTextColor(color)
    }

    private fun setClue() {
        val clue = arguments.getString(CLUE_ID_KEY)
        if (clue === "") {
            tvNextLevelClue.text = "CONGRATULATIONS!"
        } else {
            tvNextLevelClue.text = clue
        }
    }

    private fun setFunFact() {
        val fact = arguments.getString(FACT_ID_KEY)
        tvFunFact.text = fact
    }

    private fun setLevelsLeft() {
        val left = arguments.getLong(LEFT_ID_KEY)
        if (left.toInt() == 0) {
            tvLevelsLeft.text = "YOU'VE FINISHED THE WHOLE PACK!"
        } else if (left.toInt() == 1) {
            tvLevelsLeft.text = "ONLY ONE LEVEL LEFT IN PACK"
        } else {
            tvLevelsLeft.text = left.toString() + " LEVELS LEFT IN PACK"
        }
    }

    private fun setExcellent() {
        val res = context.resources
        val congrats = res.getStringArray(R.array.congrats)
        val i = Random().nextInt(congrats.size - 1)
        tvExcellent.text = congrats[i]
    }
}
